Userland/Games/ColorLines/ColorLines.cpp

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "ColorLines.h"
+#include "HueFilter.h"
+#include "Marble.h"
+#include "MarbleBoard.h"
+#include <LibConfig/Client.h>
+#include <LibGUI/MessageBox.h>
+#include <LibGUI/Painter.h>
+
+ColorLines::BitmapArray ColorLines::build_marble_color_bitmaps()
+{
+    auto marble_bitmap = MUST(Gfx::Bitmap::try_load_from_file("/res/icons/colorlines/colorlines.png"sv));
+    float const hue_degrees[Marble::number_of_colors] = {
+        0,   // Red
+        45,  // Brown/Yellow
+        90,  // Green
+        180, // Cyan
+        225, // Blue
+        300  // Purple
+    };
+    BitmapArray colored_bitmaps;
+    colored_bitmaps.ensure_capacity(Marble::number_of_colors);
+    for (int i = 0; i < Marble::number_of_colors; ++i) {
+        auto bitmap = MUST(marble_bitmap->clone());
+        HueFilter filter { hue_degrees[i] };
+        filter.apply(*bitmap, bitmap->rect(), *marble_bitmap, marble_bitmap->rect());
+        colored_bitmaps.append(bitmap);
+    }
+    return colored_bitmaps;
+}
+
+ColorLines::BitmapArray ColorLines::build_marble_trace_bitmaps()
+{
+    auto trace_bitmap = MUST(Gfx::Bitmap::try_load_from_file("/res/emoji/U+1F43E.png"sv));
```

Comment:
It _would_ be cool if we can load the emoji image directly from specifying the code point itself, rather than it's representation, but it's not the end of the world... just simply place a comment showing what emoji this is :^)